### PR TITLE
allow setting active radio-filter choice dynamically

### PR DESCRIPTION
**Pull request by HassenPy**
Allow similar behavior to react's value prop for the select tag, this will allow binding the activeFilter state to an external parent component state.
 https://facebook.github.io/react/docs/forms.html#the-select-tag

### CODE REVIEW

**Comment by HassenPy** [comment]
sorry for this, doing python few minutes before writing this lol.
if you're wondering why i need this, i'm making an adaptive filter component, shows a select on small screens and radio-filter on larger ones, i was able to synchronize the state from the radio-filter to the select element, but wasn't able to do backwards, why i need the synchronization is in case of screen swap by user from portrait <-> landscape

**Comment by alanmoo** [comment]
Thanks! Looking forward to seeing this get utilized!

**Comment by HassenPy** [comment]
Great, expect to find one by the time u get back on monday.

**Comment by arush0311** [comment]
Woudn't initial state also consider props.value attribute?

**Comment by arush0311** [comment]
we could keep both initialValue and value for backwards compatibility. Just that setting value will make it a controlled component.

**Comment by arush0311** [comment]
Ah nm. will leave a comment there.

File: src/components/radio-filter/radio-filter.jsx [hunk 1]
```diff
@@ -23,6 +24,13 @@ export default React.createClass({
 
     this.props.onChange(choice);
   },
+  componentWillReceiveProps(nextProps){
+    if (nextProps.value !== this.state.activeFilter){
+      this.setState({
+        activeFilter: nextProps.value
+      });
+    }
+  },
```
Comment (arush0311) [inline src/components/radio-filter/radio-filter.jsx]: orrect me if i am wrong, this function will execute only when we change props without changing the state of parent component, right?

Comment (HassenPy) [inline src/components/radio-filter/radio-filter.jsx]: It executes when the parent component changes a state that's passed to this component.

Comment (arush0311) [inline src/components/radio-filter/radio-filter.jsx]: Hm nm then.

Comment (arush0311) [inline src/components/radio-filter/radio-filter.jsx]: Actually this doesnot seem correct see http://stackoverflow.com/questions/32414308/updating-state-on-props-change-in-react-form

**Comment by arush0311** [comment]
@HassenPy lets wait on this until we resolve https://github.com/mozilla/mofo-ui/pull/58

**Pull request merged**